src/c++/perf_analyzer/genai-perf/genai_perf/parser.py

```diff
@@ -241,6 +267,18 @@ def _convert_str_to_enum_entry(args, option, enum):
     return args
 
 
+### Types ###
+
+
+def file_or_directory(path: str) -> Tuple[Path, str]:
+    if os.path.isfile(path):
+        return (Path(path), "file")
```

Comment:
Nice approach! I think the second field (file, directory, etc.) should be an enum. Beyond common practice and efficiency, it's easier to keep track of enums. We don't want to be doing string comparison. It'd be too easy to make/overlook mistakes.